go/database/mpt/io/live.go

```diff
@@ -74,6 +74,32 @@ func Export(ctx context.Context, directory string, out io.Writer) error {
 	}
 	defer db.Close()
 
+	return exportLive(ctx, db, out)
+}
+
+func ExportFromArchive(ctx context.Context, directory string, out io.Writer) error {
+	info, err := CheckMptDirectoryAndGetInfo(directory)
+	if err != nil {
+		return fmt.Errorf("error in input directory: %v", err)
+	}
+
+	if info.Config.Name != mpt.S5ArchiveConfig.Name {
+		return fmt.Errorf("can only support exportLive of S5 Archive instances, found %v in directory", info.Config.Name)
+	}
+
+	archive, err := mpt.OpenArchiveTrie(directory, info.Config, mpt.NodeCacheConfig{})
+	if err != nil {
+		return err
+	}
+
+	defer archive.Close()
+
+	block := ctx.Value("block")
```

Comment:
I thought at the beginning that it is a good idea to pass this parameter via context, but seeing the final version, I do not see any reason now, because this method can just have a parameter `block` in its header right?  I did not like the unification with other methods, where `block` was not used, but I cannot see why only this method could not have this parameter. 

If you want to have the same header with other methods to use method reference in `export.go,` yo can always wrap it there in another method. 

```
m := func(ctx context.Context, directory string) error {
    return ExportFromArchive(ctx, directory, out, block)
} 
```

Reply:
I dont really like your the func `m` you suggested. I will just call the functions inside a switch and then pass the error to the `Join`

Reply:
yes sure, it is fine. I was suggesting this solution because I thought you wanted to have all three methods with the same header (otherwise I do not understand why the `block` param was added to other methods), but sure calling all three methods independently in a switch statement is the most straightforward and readable approach.